assets/wizards/engagement/components/types.ts

```diff
@@ -0,0 +1,212 @@
+			from_name: string;
+			from_email: string;
+			reply_to_email: string;
+			status: string;
+		};
+	};
+	sender_name?: string;
+	sender_email_address?: string;
+	contact_email_address?: string;
+};
+
+// Props for the Prequisite component.
+export type PrequisiteProps = {
+	config: Config;
+	getSharedProps: (
+		configKey: string,
+		type: string
+	) => {
+		onChange: ( value: string | boolean ) => void;
+		disabled?: boolean;
+		checked?: boolean;
+		value?: string;
+	};
+	inFlight: boolean;
+	saveConfig: ( config: Config ) => void;
+
+	// Schema for prequisite object is defined in PHP class Reader_Activation::get_prerequisites_status().
+	prerequisite: {
+		active: boolean;
+		label: string;
+		description: string;
+		help_url: string;
+		fields?: {
+			[ fieldName: string ]: {
+				label: string;
+				description: string;
+			};
+		};
+		href?: string;
+		action_text?: string;
+		action_enabled?: boolean;
+		disabled_text?: string;
+	};
+};
+
+export type InputField = {
+	name: string;
+	type: string;
+	label: string;
+	description: string;
+	required?: boolean;
+	max_length?: number;
+	default: string | number | boolean;
+	value?: string | number | boolean;
+	options?: {
+		label: string;
+		value: string | number;
+	};
+};
+
+// Schema is defined in Newspack Campaigns: https://github.com/Automattic/newspack-popups/blob/master/includes/schemas/class-prompts.php
+export type PromptType = {
+	status: string;
+	slug: string;
+	title: string;
+	content: string;
+	featured_image_id?: number;
+	options: PromptOptions;
+	user_input_fields: [ InputField ];
+};
+
+export type PromptOptions = {
+	background_color: string;
+	display_title: boolean;
+	hide_border: boolean;
+	large_border: boolean;
+	frequency: string;
+	frequency_max: number;
+	frequency_start: number;
+	frequency_between: number;
+	frequency_reset: string;
+	overlay_color: string;
+	overlay_opacity: number;
+	overlay_size: string;
+	no_overlay_background: boolean;
+	placement: string;
+	trigger_type: string;
+	trigger_delay: number;
+	trigger_scroll_progress: number;
+	trigger_blocks_count: number;
+	archive_insertion_posts_count: number;
+	archive_insertion_is_repeating: false;
+	selected_segment_id: string;
+	post_types: Array< string >;
+	archive_page_types: Array< string >;
+	additional_classes: string;
+	excluded_categories: [
+		{
+			id: number;
+			name: string;
+		}
+	];
+	excluded_tags: [
+		{
+			id: number;
+			name: string;
+		}
+	];
+	categories: [
+		{
+			id: number;
+			name: string;
+		}
+	];
+	tags: [
+		{
+			id: number;
+			name: string;
+		}
+	];
+	campaign_groups: [
+		{
+			id: number;
+			name: string;
+		}
+	];
+};
+
+export type Attachment = {
+	source_url?: string;
+	url: string;
+};
+
+export type InputValues = {
+	[ fieldName: string ]: string | number | Array< string > | Array< number > | boolean;
+};
+
+// Props for the Prompt component.
+export type PromptProps = {
+	inFlight: boolean;
+	setInFlight: ( inFlight: boolean ) => void;
+	prompt: Prompt;
```

Comment:
`Prompt` definition is missing here. Should this be a composition of `PromptType` and `PromptOptions`?